app/ports/management/commands/update-portinfo.py

```diff
@@ -44,6 +44,12 @@ def handle(self, *args, **options):
                     found_ports_tree[portdir] = {}
                 found_ports_tree[portdir][portname] = True
 
+        # When entire portdir is removed, it won't be picked up while looping over portindex.json
```

Comment:
I find the name of function `handle` somewhat non-intuitive, as well as its description somewhat not reflecting the exact purpose.

Could we replace `found_ports_tree` in this function and `dict_of_portdirs` inside `mark_deleted` with a variable using the same name at both places? Maybe `dict_of_portdirs_with_ports` (`dict_of_portdirs_with_set_of_ports`). That's not the best name either, but maybe you can think of something better.